File: 백준/Silver/10773. 제로/제로.py
import sys

k = int(sys.stdin.readline())
writeList = []
cntt = 0

for i in range(k):
    n = int(input())

    if n == 0 and len(writeList) != 0:
        del writeList[-1]
    else:
        writeList.append(n)

print(sum(writeList))

File: 백준/Silver/10773. 제로/제로.kt
fun main() {
    val k : Int = readLine()!!.toInt()
    val writeList = mutableListOf<Int>()

    repeat(k) {
        val n : Int = readLine()!!.toInt()

        if (n == 0 && writeList.isNotEmpty()) {
            writeList.removeAt(writeList.size-1)
        } else {
            writeList.add(n)
        }
    }
    print(writeList.sum())
}